op-node/rollup/derive/channel_in_reader.go

```diff
@@ -86,15 +88,26 @@ func (cr *ChannelInReader) NextBatch(ctx context.Context) (Batch, error) {
 		cr.NextChannel()
 		return nil, NotEnoughData
 	}
-	if batchData.BatchType == SingularBatchType {
-		return &batchData.SingularBatch, nil
-	} else {
+	switch batchData.GetBatchType() {
```

Comment:
This change will conflict with diff introduced in https://github.com/ethereum-optimism/optimism/pull/7621. Rebase needed.